hub-auth-js/app/Main.js

```diff
@@ -0,0 +1,139 @@
+  "ApplicationBase/support/itemUtils",
+  "ApplicationBase/support/domHelper",
+  "esri/identity/OAuthInfo",
+  "esri/identity/IdentityManager",
+  "esri/request",
+  "dojo/dom",
+  "dojo/on"
+], function (
+  declare,
+  ApplicationBase,
+  i18n,
+  itemUtils,
+  domHelper,
+  OAuthInfo,
+  esriId,
+  esriRequest,
+  dom,
+  on
+) {
+    return declare(null, {
+
+      constructor: function () {
+        this.CSS = {
+          loading: "configurable-application--loading"
+        };
+        this.base = null;
+      },
+
+      init: function (base) {
+        if (!base) {
+          console.error("ApplicationBase is not defined");
+          return;
+        }
+        domHelper.setPageLocale(base.locale);
+        domHelper.setPageDirection(base.direction);
+
+        this.base = base;
+        var config = base.config, results = base.results;
+        var find = config.find, marker = config.marker;
+        var webMapItems = results.webMapItems;
+        var validWebMapItems = webMapItems.map(function (response) {
+          return response.value;
+        });
+        var firstItem = validWebMapItems[0];
+        if (!firstItem) {
+          console.error("Could not load an item to display");
+          return;
+        }
+        config.title = !config.title ? itemUtils.getItemTitle(firstItem) : null;
+        domHelper.setPageTitle(config.title);
+
+        var portalItem = this.base.results.applicationItem.value;
+        var appProxies = (portalItem && portalItem.appProxies) ? portalItem.appProxies : null;
+        var viewContainerNode = document.getElementById("viewContainer");
+        var defaultViewProperties = itemUtils.getConfigViewProperties(config);
+        validWebMapItems.forEach(function (item) {
+
+          var viewNode = document.createElement("div");
+          viewContainerNode.appendChild(viewNode);
+
+          var viewProperties = defaultViewProperties;
+          viewProperties.container = viewNode;
+          itemUtils.createMapFromItem({ item: item, appProxies: appProxies })
+            .then(function (map) {
+              viewProperties.map = map;
+              return itemUtils.createView(viewProperties)
+                .then(function (view) {
+                  return itemUtils.findQuery(find, view)
+                    .then(function () { return itemUtils.goToMarker(marker, view); });
+                });
+            });
+        });
+        document.body.classList.remove(this.CSS.loading);
+
+        /*
+        everything up to here is exactly the same as the 'webmap-js' sample in this repository
+        everything afterward is lifted from our 3.x JSAPI OAuth sample
+        https://developers.arcgis.com/javascript/3/jssamples/portal_oauth_popup.html
+        */
+
+        // if the url of the community org is known, there is no need to fetch it via a portals/self call
+        if (!config.communityPortalUrl) {
+          esriRequest(config.portalUrl + '/sharing/rest/portals/self?f=json').then(
```

Comment:
I don't think we need to do a esriRequest here. `base.portal` should already have this I think. Or at least `base.portal.load.then()`